src/metrics/mongoose_metrics.erl

```diff
@@ -64,27 +66,17 @@
 -spec init() -> ok.
 init() ->
     prepare_prefixes(),
-    create_global_metrics(),
-    lists:foreach(
-        fun(HostType) ->
-            init_predefined_host_type_metrics(HostType)
-        end, ?ALL_HOST_TYPES),
+    create_vm_metrics(),
+    create_global_metrics(?GLOBAL_COUNTERS),
+    create_data_metrics(),
+    create_host_type_metrics(),
     init_subscriptions().
 
-create_global_metrics() ->
-    lists:foreach(fun({Metric, FunSpec, DataPoints}) ->
-        FunSpecTuple = list_to_tuple(FunSpec ++ [DataPoints]),
-        catch ensure_metric(global, Metric, FunSpecTuple)
-    end, ?VM_STATS),
-    lists:foreach(fun({Metric, Spec}) -> ensure_metric(global, Metric, Spec) end,
-                  ?GLOBAL_COUNTERS),
-    create_data_metrics().
-
--spec init_predefined_host_type_metrics(mongooseim:host_type()) -> ok.
-init_predefined_host_type_metrics(HostType) ->
-    create_metrics(HostType),
-    Hooks = mongoose_metrics_hooks:get_hooks(HostType),
-    gen_hook:add_handlers(Hooks).
+-spec init_mongooseim_metrics() -> ok.
+init_mongooseim_metrics() ->
+    create_host_type_hook_metrics(),
+    create_global_metrics(?MNESIA_COUNTERS),
+    ok.
```

Comment:
The function above already returns ok.

Reply:
All right, fixed but also squashed that commit because such a small change is too dirty to have its own separate commit 😁